lib/os/crc8_sw.c

```diff
@@ -24,3 +25,32 @@ u8_t crc8_ccitt(u8_t val, const void *buf, size_t cnt)
 	}
 	return val;
 }
+
+u8_t crc8(const u8_t *src, size_t len, u8_t polynomial, u8_t initial_value,
```

Comment:
Is this code original, or did you get it from a public domain source?

Reply:
The code is original but modelled after a pseudo code in the net, probably all the code snippets in the net look quite similar. The code will be used in GSM muxing that I will send soon. Why do you ask?

Reply:
This function is in practical sense identical to the first crc16() one that was created from scratch, and both of them are approximately the one and only non-table driven algorithmic way to implement said CRC functionality.